code/modules/mob/living/carbon/human/ai/action_datums/treat_ally.dm

```diff
@@ -0,0 +1,54 @@
+/datum/ai_action/treat_ally
+	name = "Treat Ally"
+	action_flags = ACTION_USING_HANDS | ACTION_USING_LEGS
+
+/datum/ai_action/treat_ally/get_weight(datum/human_ai_brain/brain)
+	if(!brain.treat_allies)
+		return 0
+
+	if(!brain.found_injured_ally)
+		return 0
+
+	if(brain.healing_someone)
+		return 0
+
+	var/should_fire_offscreen = (brain.target_turf && !COOLDOWN_FINISHED(brain, fire_offscreen))
```

Comment:
So does the fire_offscreen cooldown work as a sort of "alert state", where the AI tries to fire offscreen until the cooldown ends?

Reply:
Yeah, also they are trying to shoot at last known enemy position, which is determined when AI gets hit by an enemy bullet, so we simulate a suppressive/counter/anti-cheese fire for the duration of this timer